src/Illuminate/Support/HigherOrderCollectionProxy.php

```diff
@@ -54,7 +54,15 @@ public function __get($key)
     public function __call($method, $parameters)
     {
         return $this->collection->{$this->method}(function ($value) use ($method, $parameters) {
-            return $value->{$method}(...$parameters);
+            if (is_object($value)) {
+                return $value->{$method}(...$parameters);
+            }
+
+            if ($method == 'empty') {
```

Comment:
No need for this hack, just have:

```php
if (is_object($value)) {
    return $value->{$method}(...$parameters);
}

return $method(...array_merge([$value], $parameters));
```

A quick test that this behaves as we want: https://3v4l.org/f1hC8.